File: elixir/resistor-color-duo/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value([c1, c2 | _]) do
    bands = [:black, :brown, :red, :orange, :yellow, :green, :blue, :violet, :grey, :white]
    Enum.find_index(bands, fn x -> x == c1 end) * 10 +
    Enum.find_index(bands, fn x -> x == c2 end)
  end
end

File: typescript/resistor-color-duo/resistor-color-duo.ts
import { inspect } from "util"
let colours: Array<string> = 
  ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']

export function decodedValue(codes: Array<string>) {
  // throw new Error('Remove this statement and implement this function')
  // for (var code of codes) {
  //   console.log(colours.indexOf(code))
  // }
  // (10).toString()
  var code = 
    colours.indexOf(codes[0]) * 10 + 
    colours.indexOf(codes[1])
    
  return code;
}
